File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Events/AtEndOfTurnPreEndTurnCardsMessage.ts
import { IMessage } from "../../../../Core/ObserverMessageKit";
import AbstractCreature from "../Unit/AbstractCreature";

export class AtEndOfTurnPreEndTurnCardsMessage implements IMessage{
    public static readonly Id = "AtEndOfTurnPreEndTurnCards";
    public constructor(public unit:AbstractCreature){

    }
}

File: Unity/Assets/Scripts/ModTheSpire/Event/Ids/AtEndOfTurnPreEndTurnCardsMessage.cs
using ModTheSpire.Unit;

namespace ModTheSpire.Event.Ids
{
    ///回合结束弃牌
    public class AtEndOfTurnPreEndTurnCardsMessage : IEventHandle
    {
        public AbstractCreature Unit;
        public AtEndOfTurnPreEndTurnCardsMessage(AbstractCreature unit)
        {
            Unit = unit;
        }
    }
}